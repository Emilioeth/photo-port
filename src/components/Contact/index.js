import React, { useState } from 'react';
import { validateEmail } from '../../utils/helpers';

function ContactForm() {

    const [formState, setFormState] = useState({ name: '', email: '', message: '' });
    const { name, email, message } = formState;

    const [errorMessage, setErrorMessage] = useState('');

    function handleChange(e) {
        if (e.target.name === 'email') {
            const isValid = validateEmail(e.target.value);
            // isValid conditional statement
            if (!isValid) {
                setErrorMessage('Your email is invalid.');
            } else {
                if (!e.target.value.length) {
                  setErrorMessage(`${e.target.name} is required.`);
                } else {
                  setErrorMessage('');
                }
            }
          } 
          if (!errorMessage) {
            setFormState({ ...formState, [e.target.name]: e.target.value });
          }
      }

      function handleSubmit(e) {
        e.preventDefault();
        console.log(formState);
      }
      
      console.log(formState);

    return (
        <section>
          <h1 data-testid="contact-title" >Contact me</h1>
          <form id="contact-form" onSubmit={handleSubmit}>
            <div>
              <label htmlFor="name">Name:</label>
              <input type="text" defaultValue={name} onBlur={handleChange} name="name" />
            </div>
            <div>
              <label htmlFor="email">Email address:</label>
              <input type="email" defaultValue={email} onBlur={handleChange} name="email" />
            </div>
            <div>
              <label htmlFor="message">Message:</label>
              <textarea name="message" defaultValue={message} onBlur={handleChange} rows="5"  />
            </div>
            <button type="submit" data-testid="contact-btn">Submit</button>
          </form>
          {errorMessage && (
            <div>
                <p className="error-text">{errorMessage}</p>
            </div>
            )}
        </section>
    );

}
    
export default ContactForm