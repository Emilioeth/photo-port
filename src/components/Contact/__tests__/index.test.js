import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Contact from '..'

afterEach(cleanup)

describe('Contact is rendering', () => {

    it('renders', () => {
      render(<Contact />);
    });
  
    it('matches snapshot', () => {
      const { asFragment } = render(<Contact />)
      expect(asFragment()).toMatchSnapshot()
    })

    it('renders contact title', () => {
        const { getByTestId } = render(<Contact />)
        expect(getByTestId('contact-title')).toHaveTextContent('Contact')
    })

    it('renders contact btn', () => {
        const { getByTestId } = render(<Contact />)
        expect(getByTestId('contact-btn')).toHaveTextContent('Submit')
    })

  })