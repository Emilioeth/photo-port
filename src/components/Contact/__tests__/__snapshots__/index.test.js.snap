// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact is rendering matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="contact-title"
    >
      Contact me
    </h1>
    <form
      id="contact-form"
    >
      <div>
        <label
          for="name"
        >
          Name:
        </label>
        <input
          name="name"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email address:
        </label>
        <input
          name="email"
          type="email"
          value=""
        />
      </div>
      <div>
        <label
          for="message"
        >
          Message:
        </label>
        <textarea
          name="message"
          rows="5"
        />
      </div>
      <button
        data-testid="contact-btn"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
